Initial attempt to add scroll to top button

justincalleja opened the pull request:
This is for issue #41. There are issues with this pull request (just sharing).

Currently, when "scroll to top" button is clicked, the data-tooltip does not go away until you focus on something else.

I have added commented out code in _tooltip.scss but it's not working right with it uncommented either (the issue then becomes that, although the data-tooltip goes away after clicking, you cannot click / hover on the "scroll to top" button again and get the data-tooltip - not without first focusing on something else).

LouisBarranqueiro commented:
Hi man, try to fix this correctly. I'm currently working on other tasks :)

justincalleja commented:
Hey Louis, I think this commit fixes it. I have removed the following where they were being used in my case (.tooltip and .tooltip--top):

```css
&:focus:before,
&:focus:after
```

Left them there for .tooltip--left, .tooltip-bottom, and .tooltip--right as I don't know if it breaks anything else / not too comfortable with CSS. Basically, I wanted the transition and it being visible only on hover (the on focus was keeping it there after clicking).

justincalleja commented:
On other thing. I have only tested on desktop. If you minimize the viewport enough, the scroll to top button gets pushed down. This also happens for other buttons (prev/next, and the social links) if the viewport gets small enough.

Is this a known issue? Should the bottom links become an expandable menu when it's small enough? (or maybe some other way).

LouisBarranqueiro commented:
Normally, you don't have to edit tooltips css components. I thought about this features and I believe that the best way (UI & UX) to do that is to add at the right before comment button the `list` font-awesome icon. The icon is beautiful :D
![tables-of-content](https://cloud.githubusercontent.com/assets/4705626/8594402/99e11d06-2640-11e5-93bc-7993ae77e041.png)

What do you think? By adding this new button, you have to greatly edit `_post-actions.scss` file to handle the behavior on all medias. Currently the button falls down when you resize the screen instead of disappear when there is no more space. Will you be able to do that? :)

justincalleja commented:
I will give it a go. Is there going to be something which replaces them when the screen gets too small? To pull up a menu perhaps? (Maybe "prev" and "next" and "scroll to top" can stay there and the menu can be for the social buttons).

LouisBarranqueiro commented:
You're right.  i will implement this feature -> on small screen all share buttons are grouped in only one with like this : 
![capture d ecran 2015-07-10 a 13 31 34](https://cloud.githubusercontent.com/assets/4705626/8617835/0dd1aaec-2708-11e5-8c98-309f92c45bef.png)

Are you ok?

justincalleja commented:
Definitely :+1:  Thanks!

LouisBarranqueiro commented:
Cool :) I will implement this share buttons feature this WE. :) After that, you will implement top button feature. :)

justincalleja commented:
Sure no problem. I'm guessing that now there won't be a space issue no? If it's just adding a button I can do that tomorrow. 

The styles to hide the button and lose it completely... that I'd have to look around a bit at what media queries to use. (Though if that ends up being the case, I would probably prefer having it in the menu even though it's not a social media button).

justincalleja commented:
I removed the previous button. Now I added the "fa-list" as per your screen shot above. There is no tool tip for it.

Running [pageres](https://github.com/sindresorhus/pageres) at '480x320', '1024x768', and 'iphone 5s' gives the new button being pushed down at '480x320' and 'iphone 5s' (320x568).

Do we need to add media queries given that we're going for a menu? If so, do I just remove the fa-list with no means of accessing it?


Cheers,
J

LouisBarranqueiro commented:
No, we don't need to add media queries because there will be 3 buttons as is currently the case. Can you pull request on `dev` branch please and it will be perfect :)

justincalleja commented:
Ah yes `dev`... I knew I read that somewhere and forgot about it.... The only way I know how to do this is closing this PR then. Will open a new one.
